tools/pipelines/build-client.yml

```diff
@@ -178,28 +178,32 @@ variables:
 extends:
   template: /tools/pipelines/templates/build-npm-client-package.yml@self
   parameters:
+    ${{ if eq(variables['System.TeamProject'], 'internal') }}:
+      # Overriding some variables should only be possible in the internal project
+      releaseBuildOverride: ${{ parameters.releaseBuildOverride }}
+      publishOverride: ${{ parameters.publishOverride }}
+      packageTypesOverride: ${{ parameters.packageTypesOverride }}
```

Comment:
Since we don't want to ever release from the public project, now overriding those parameters only has an effect in the internal project. Might be cleaner if we had two main pipeline files, one for public and one for internal, each tied to the pipeline in the corresponding project. The logic would still be shared in `build-npm-client-package.yml`.